packages/babel-plugin-jsx-dom-expressions/test/__dom_fixtures__/conditionalExpressions/output.js

```diff
@@ -52,13 +52,12 @@ const template7 = (() => {
     _el$7,
     (() => {
       var _c$3 = _$memo(() => state.count > 5);
-      return () =>
-        _c$3()
-          ? (() => {
-              var _c$4 = _$memo(() => !!state.dynamic);
-              return () => (_c$4() ? best : good());
-            })()
-          : bad;
+      return _c$3()
```

Comment:
I tested this component (template7 in conditionalExpressions, and despite a changed output, it still render how i would expect it to.

Reply:
Actually no, it gives a hydration error - so that's an unwanted side-effect.

Reply:
Turns out that hydration error didn't come from this change but was there all-ready ([StackBlitz](https://stackblitz.com/edit/github-7dcnxd-sgpkre?file=src%2Fapp.tsx))